bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/wrapper/VehiclePositionWrapper.java

```diff
@@ -69,19 +73,16 @@ public State getPosition() {
     }
 
     public State isCalculated() {
-        return vehicle.calculatedPosition.latitude != null ? OnOffType.ON : OnOffType.OFF;
+        return position.map(pos -> isCalculated ? (State) OnOffType.ON : OnOffType.OFF).orElse(UnDefType.UNDEF);
     }
 
     public State isHeading() {
-        return (vehicle.position.isHeading() || vehicle.calculatedPosition.isHeading()) ? OnOffType.ON : OnOffType.OFF;
+        return position.map(pos -> pos.isHeading() ? (State) OnOffType.ON : OnOffType.OFF).orElse(UnDefType.UNDEF);
     }
 
     public State getTimestamp() {
-        return vehicle.position.getTimestamp().isPresent()
-                ? new DateTimeType(vehicle.position.getTimestamp().get().withZoneSameInstant(ZoneId.systemDefault()))
-                : vehicle.calculatedPosition.getTimestamp().isPresent()
-                        ? new DateTimeType(vehicle.calculatedPosition.getTimestamp().get()
-                                .withZoneSameInstant(ZoneId.systemDefault()))
-                        : UnDefType.NULL;
+        return position.map(pos -> pos.getTimestamp()
+                .map(dt -> (State) new DateTimeType(dt.withZoneSameInstant(ZoneId.systemDefault())))
+                .orElse(UnDefType.NULL)).orElse(UnDefType.UNDEF);
```

Comment:
```suggestion
        return position.flatMap(pos -> pos.getTimestamp())
                .map(dt -> (State) new DateTimeType(dt.withZoneSameInstant(ZoneId.systemDefault())))
                .orElse(UnDefType.NULL);
```

Reply:
You can ignore this suggestion if it isn't the behavior you want.

Reply:
Yes, this is obviously better.